File: 2717-semi-ordered-permutation/2717-semi-ordered-permutation.java
class Solution {
    public int semiOrderedPermutation(int[] nums) {
        int minIndex = -1;
        int maxIndex = -1;
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        for(int i = 0;i<nums.length;i++){
            if(nums[i] < min){
                min = nums[i];
                minIndex = i;
            }
            if(nums[i] > max){
                max = nums[i];
                maxIndex = i;
            }
        }
        int totalSwaps = minIndex + (nums.length - maxIndex - 1);
        return maxIndex > minIndex ? totalSwaps : totalSwaps - 1;
    }
}

File: 2717-semi-ordered-permutation/2717-semi-ordered-permutation.py
class Solution:
    def semiOrderedPermutation(self, nums: List[int]) -> int:
        minIndex = -1
        maxIndex = -1
        Min = float('inf')
        Max = float('-inf')
        for i,num in enumerate(nums):
            if num < Min:
                Min = num
                minIndex = i
            if num > Max:
                Max = num 
                maxIndex = i
        totalSwaps = minIndex + (len(nums) - maxIndex - 1)
        return totalSwaps if maxIndex > minIndex else totalSwaps - 1